backend/src/packages/drivers/driver.service.ts

```diff
@@ -0,0 +1,157 @@
+    this.groupService = groupService;
+  }
+
+  public async findById(id: number): Promise<DriverEntityT | null> {
+    const [driver = null] = await this.driverRepository.find({ id });
+
+    return driver ? DriverEntity.initialize(driver).toObject() : null;
+  }
+
+  public async findAllByBusinessId(
+    businessId: number,
+  ): Promise<DriverGetAllResponseDto> {
+    const items = await this.driverRepository.findAllByBusinessId(businessId);
+
+    return {
+      items: items.map((it) => it.toObject()),
+    };
+  }
+
+  public async create({
+    payload,
+    id,
+  }: DriverCreatePayload): Promise<DriverAddResponseWithGroup> {
+    const { password, email, lastName, firstName, phone, driverLicenseNumber } =
+      payload;
+
+    const { result: doesDriverExist } = await this.driverRepository.checkExists(
+      {
+        driverLicenseNumber,
+      },
+    );
+
+    if (doesDriverExist) {
+      throw new HttpError({
+        status: HttpCode.BAD_REQUEST,
+        message: HttpMessage.DRIVER_ALREADY_EXISTS,
+      });
+    }
+    const group = await this.groupService.findByKey(UserGroupKey.DRIVER);
+
+    if (!group) {
+      throw new HttpError({
+        message: HttpMessage.INVALID_GROUP,
+        status: HttpCode.BAD_REQUEST,
+      });
+    }
+
+    const user = await this.userService.create({
+      password,
+      email,
+      lastName,
+      firstName,
+      phone,
+      groupId: group.id,
+    });
+
+    const driver = await this.driverRepository.create(
+      DriverEntity.initializeNew({
+        driverLicenseNumber,
+        businessId: id,
+        userId: user.id,
+      }),
+    );
+
+    const driverObject = driver.toObject();
+
+    return { ...user, group, driver: driverObject };
+  }
+
+  public async update({
+    driverId,
+    payload,
+  }: DriverUpdatePayload): Promise<DriverUpdateResponseDto> {
+    const foundDriver = await this.findById(driverId);
+
+    if (!foundDriver) {
+      throw new NotFoundError({});
+    }
+
+    const { result: doesDriverExist } = await this.driverRepository.checkExists(
+      {
+        driverLicenseNumber: payload.driverLicenseNumber,
+      },
+    );
+
+    if (doesDriverExist) {
+      throw new HttpError({
+        status: HttpCode.BAD_REQUEST,
+        message: HttpMessage.LICENSE_NUMBER_ALREADY_EXISTS,
+      });
+    }
+
+    const user = await this.userService.update(driverId, payload);
+
+    const driver = await this.driverRepository.update({
+      id: foundDriver.id,
+      payload,
+    });
+
+    const driverObject = driver.toObject();
+
+    return { ...user, driver: driverObject };
+  }
+
+  public async delete(id: number): Promise<boolean> {
+    const foundDriver = await this.findById(id);
+
+    if (!foundDriver) {
+      throw new NotFoundError({});
+    }
+
+    const isDriverDeleted = await this.driverRepository.delete(id);
+
+    if (isDriverDeleted) {
+      return await this.userService.delete(foundDriver.userId);
+    } else {
```

Comment:
Maybe we should use soft delete for this, because if we delete driver,  we could not see his statistics

Reply:
I agree, soft deletion is more appropriate here. But for now, I suggest leaving it as it is. Because in addition to changes to the driver repository, we need to do some work in the user repository and the user service. And this task already has too much code. I think it's better to put the implementation of soft delete in a separate task.